Assets/Shopify/UIToolkit/Editor/ProductPicker.cs

```diff
@@ -0,0 +1,155 @@
+﻿namespace Shopify.UIToolkit.Editor {
+    using System.Collections;
+    using System.Collections.Generic;
+    using UnityEngine;
+    using UnityEditor;
+    using Shopify.Unity;
+    using Shopify.Unity.SDK;
+    using System;
+    using System.Linq;
+
+    public interface IProductPickerView {
+        void DrawProductLoadingState(SerializedObject serializedObject);
+        void DrawProductPicker(SerializedObject serializedObject);
+        void DrawProductLoadingError(SerializedObject serializedObject);
+        void DrawShopHasNoProductsState(SerializedObject serializedObject);
+    }
+
+    public class ProductPicker : IProductPickerView {
+        private class ProductNameToGIDMapEntry {
```

Comment:
Nit: Does it ever make sense to have a null `ProductNameToGIDMapEntry`? Might make more sense to use value types here by declaring it as a `struct` instead of `class`.